File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/parts/resource/ResourceBars.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.state.RunningState;
import com.bonepl.razersdk.animation.IFrame;
import com.bonepl.razersdk.sdk.RzKey;
import com.bonepl.razersdk.sdk.RzKeySelector;

import java.util.Collections;
import java.util.List;
import java.util.Set;

import static com.bonepl.razersdk.sdk.RzKey.RZKEY_BACKSPACE;
import static com.bonepl.razersdk.sdk.RzKey.RZKEY_TILDE;

public final class ResourceBars {
    private static final List<RzKey> RESOURCE_BAR_KEYS = new RzKeySelector()
            .withRowOf(RZKEY_TILDE)
            .withColumnBetween(RZKEY_TILDE, RZKEY_BACKSPACE)
            .sortedByColumn()
            .asList();

    private static final Set<String> NO_MANA_BAR_CHAMPIONS
            = Set.of("Aatrox", "Dr. Mundo", "Garen", "Katarina", "Riven", "Viego", "Yone", "Zac");
    private static final Set<String> ENERGY_BAR_CHAMPIONS
            = Set.of("Akali", "Kennen", "Lee Sin", "Shen", "Zed");

    private ResourceBars() {
    }

    public static IFrame getResourceBarForActivePlayerChampion() {
        String activePlayerChampionName = RunningState.getGameState().getPlayerList().getActivePlayer().championName();
        if (NO_MANA_BAR_CHAMPIONS.contains(activePlayerChampionName)) {
            return new NoResourceBar();
        }

        if (ENERGY_BAR_CHAMPIONS.contains(activePlayerChampionName)) {
            return new EnergyBar();
        }

        return switch (activePlayerChampionName) {
            case "Gnar" -> new GnarFuryBar();
            case "Kled" -> new KledCourageBar();
            case "Mordekaiser" -> new MordekaiserShieldBar();
            case "Renekton" -> new RenektonFuryBar();
            case "Rengar" -> new RengarFerocityBar();
            case "Rek'Sai", "Tryndamere" -> new RedFuryBar();
            case "Rumble" -> new RumbleHeatBar();
            case "Sett" -> new SettGritBar();
            case "Shyvana" -> new ShyvanaDragonFuryBar();
            case "Vladimir" -> new VladimirBloodPoolBar();
            case "Yasuo" -> new YasuoWindBar();
            case "Yone" -> new YoneCloneBar();
            default -> new ManaBar();
        };
    }

    public static List<RzKey> getResourceBarKeys() {
        return Collections.unmodifiableList(RESOURCE_BAR_KEYS);
    }

    public static Set<String> getEnergyBarChampions() {
        return ENERGY_BAR_CHAMPIONS;
    }
}

File: src/hud/parts/resource/ResourceBars.ts
import {IFrame} from '../../../razer-sdk/animation/IFrame.js';
import {RzKey} from '../../../razer-sdk/sdk/RzKey.js';
import {RzKeySelector} from '../../../razer-sdk/sdk/RzKeySelector.js';
import {RunningState} from '../../../state/RunningState.js';
import {BelVethLavenderBar} from './BelVethLavenderBar.js';
import {EnergyBar} from './EnergyBar.js';
import {GnarFuryBar} from './GnarFuryBar.js';
import {KledCourageBar} from './KledCourageBar.js';
import {ManaBar} from './ManaBar.js';
import {MordekaiserShieldBar} from './MordekaiserShieldBar.js';
import {NoResourceBar} from './NoResourceBar.js';
import {RedFuryBar} from './RedFuryBar.js';
import {RenektonFuryBar} from './RenektonFuryBar.js';
import {RengarFerocityBar} from './RengarFerocityBar.js';
import {RumbleHeatBar} from './RumbleHeatBar.js';
import {SettGritBar} from './SettGritBar.js';
import {ShyvanaDragonFuryBar} from './ShyvanaDragonFuryBar.js';
import {VladimirBloodPoolBar} from './VladimirBloodPoolBar.js';
import {YasuoWindBar} from './YasuoWindBar.js';
import {YoneCloneBar} from './YoneCloneBar.js';

const RESOURCE_BAR_KEYS: RzKey[] = new RzKeySelector()
    .withRowOf(RzKey.RZKEY_TILDE)
    .withColumnBetween(RzKey.RZKEY_TILDE, RzKey.RZKEY_BACKSPACE)
    .sortedByColumn()
    .asList();

const NO_MANA_BAR_CHAMPIONS = new Set([
    'Aatrox', 'Briar', 'Dr. Mundo', 'Garen', 'Katarina', 'Riven', 'Viego', 'Zac',
]);

const ENERGY_BAR_CHAMPIONS = new Set([
    'Akali', 'Ambessa', 'Kennen', 'Lee Sin', 'Shen', 'Zed',
]);

export function getResourceBarForActivePlayerChampion(): IFrame {
    const gameState = RunningState.getGameState();
    if (!gameState.playerList || !gameState.playerRiotId) return new ManaBar();
    const activePlayerChampionName = gameState.playerList
        .getActivePlayer(gameState.playerRiotId).championName;

    if (NO_MANA_BAR_CHAMPIONS.has(activePlayerChampionName)) {
        return new NoResourceBar();
    }

    if (ENERGY_BAR_CHAMPIONS.has(activePlayerChampionName)) {
        return new EnergyBar();
    }

    switch (activePlayerChampionName) {
        case "Bel'Veth": return new BelVethLavenderBar();
        case 'Gnar': return new GnarFuryBar();
        case 'Kled': return new KledCourageBar();
        case 'Mordekaiser': return new MordekaiserShieldBar();
        case 'Renekton': return new RenektonFuryBar();
        case 'Rengar': return new RengarFerocityBar();
        case "Rek'Sai":
        case 'Tryndamere': return new RedFuryBar();
        case 'Rumble': return new RumbleHeatBar();
        case 'Sett': return new SettGritBar();
        case 'Shyvana': return new ShyvanaDragonFuryBar();
        case 'Vladimir': return new VladimirBloodPoolBar();
        case 'Yasuo': return new YasuoWindBar();
        case 'Yone': return new YoneCloneBar();
        default: return new ManaBar();
    }
}

export function getResourceBarKeys(): readonly RzKey[] {
    return RESOURCE_BAR_KEYS;
}

export function getEnergyBarChampions(): ReadonlySet<string> {
    return ENERGY_BAR_CHAMPIONS;
}
